### PR TITLE
Editor: Fixed assets lost when importing zipped OBJ/MTL

### LINKED ISSUE #19781: when improt Obj/Mtl/jpg(png) .zip files using three.js/editor ,a bug occurs! map undefined
##### Description of the problem

The obj format model is a very commonly used format. When importing obj and mtl, you need zip. . After importing, the geometry and material can be loaded normally, but the texture map has an error. After checking, I found that it is related to the following code. It is not useful to process the material-related pictures in the zip. . This is not necessarily a bug, but it affects the experience more.

`
//editor/js/Loader.js( function handleZip() )
// Poly

if ( zip.files[ 'model.obj' ] && zip.files[ 'materials.mtl' ] ) {

	var materials = new MTLLoader().parse( zip.file( 'materials.mtl' ).asText() );
	var object = new OBJLoader().setMaterials( materials ).parse( zip.file( 'model.obj' ).asText() );
	editor.execute( new AddObjectCommand( editor, object ) );

}
`

##### Three.js version

- [ ] Dev
- [ ] r118
- [ ] ...

##### Browser

- [x] All of them

##### OS

- [ ] Windows

##### Hardware Requirements (graphics card, VR Device, ...)
image

### PR DESCRIPTION
**Pull request by ycw**
fix: #19781

Preview: https://raw.githack.com/ycw/three.js/editor-imports-obj-zip-with-textures/editor/index.html

Test file: [Archiv.zip](https://github.com/mrdoob/three.js/files/15217563/Archiv.zip) by https://github.com/mrdoob/three.js/issues/19781#issuecomment-653415162

https://github.com/mrdoob/three.js/assets/1063018/6928a8ed-ff85-4031-aeff-a4f0ea6a4806

### CODE REVIEW

File: examples/jsm/loaders/MTLLoader.js [hunk 1]
```diff
@@ -365,6 +365,8 @@ class MaterialCreator {
 
 			}
 
+			map.sourceFile = value; // for UITexture
```
Comment (Mugen87) [inline examples/jsm/loaders/MTLLoader.js:368]: No loader sets `sourceFile`. It is not even a property of `Texture`. 

I understand `sourceFile` is used in context of the editor but addons should not implement any sort of editor specific logic. So I don't think this change is acceptable.

Comment (Mugen87) [reply examples/jsm/loaders/MTLLoader.js:368]: Can you please explain why changes to `MTLLoader` are necessary in the first place?

Comment (ycw) [reply examples/jsm/loaders/MTLLoader.js]: OBJLoader takes MTLLoader artifacts as `materials` to generate objects, and those materials maps (if any) do not bind filename to anywhere, so the cause belong to MTLLoader. :)

File: editor/js/libs/ui.three.js [hunk 2]
```diff
@@ -200,7 +200,7 @@ class UITexture extends UISpan {
 
 			if ( image !== undefined && image !== null && image.width > 0 ) {
 
-				canvas.title = texture.sourceFile;
+				canvas.title = texture.sourceFile || texture.name;
```
Comment (Mugen87) [inline editor/js/libs/ui.three.js:203]: I have managed to fix the loading of the zip file without any changes to `ui.three.js` and `MTLLoader`. The title of the canvas can be considered as optional and ignored.

File: editor/js/Loader.js [hunk 3]
```diff
@@ -858,16 +858,36 @@ function Loader( editor ) {
...
+		const manager = new THREE.LoadingManager();
+		manager.setURLModifier( function ( url ) {
+
+			const u8a = zip[ url ];
+
+			if ( u8a ) {
+
+				console.log( 'Loading', url );
+
+				const blob = new Blob( [ u8a ], { type: 'application/octet-stream' } );
```
Comment (Mugen87) [inline editor/js/Loader.js:870]: Can you please retain the `file` variable and also `file.buffer`. I have tested it locally and there should be no need to change the URL modifier. 

It is also sufficient to just add it to `MTLLoader`.  `OBJLoader` does not need the manager.

Comment (Mugen87) [reply editor/js/Loader.js:870]: So it should be:
```js
const manager = new THREE.LoadingManager();
manager.setURLModifier( function ( url ) {

	const file = zip[ url ];

	if ( file ) {

		console.log( 'Loading', url );

		const blob = new Blob( [ file.buffer ], { type: 'application/octet-stream' } );
		return URL.createObjectURL( blob );

	}

	return url;

} );
```

Comment (ycw) [reply editor/js/Loader.js:870]: good.

File: editor/js/libs/ui.three.js (hunk 2 above)
Comment (ycw) [reply editor/js/libs/ui.three.js:203]: It will show `undefined` when users hover on the texture input, expected?

Comment (ycw) [reply editor/js/libs/ui.three.js:203]: How about `canvas.title = texture.sourceFile || ''` ? this way nothing will be shown when hovering the texture input

File: editor/js/Loader.js [hunk 4]
```diff
@@ -858,16 +858,36 @@ function Loader( editor ) {
 
 		const zip = unzipSync( new Uint8Array( contents ) );
 
+		const manager = new THREE.LoadingManager();
+		manager.setURLModifier( function ( url ) {
+
+			const u8a = zip[ url ];
```
Comment (Mugen87) [inline editor/js/Loader.js]: What is `u8a`^^? TBH, I think `file` was easier to understand.

**Comment by ycw** [comment]
done :D

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**